packages/esm-patient-immunizations-app/src/immunizations/immunizations-form.component.tsx

```diff
@@ -119,14 +149,27 @@ const ImmunizationsForm: React.FC<DefaultWorkspaceProps> = ({ patientUuid, close
     updateSingle('currentDose', currentDose);
   };
 
+  if (isLoading || vaccineState === undefined)
+    return <span className={styles.vaccineLoader}>{`${t('loadingVaccine', 'Loading Vaccine')} ...`} </span>;
+
   return (
-    <Form className={styles.form} onSubmit={handleFormSubmit} data-testid="immunization-form">
-      <div>
-        <h4 className={styles.immunizationSequenceSelect}>
-          {`${t('vaccine', 'Vaccine')} : ${formState?.vaccineName}`}{' '}
-        </h4>
+    <Form className={styles.form} onSubmit={handleFormSubmit}>
+      <Stack gap={4}>
+        {vaccines?.length ? (
+          <section className={styles.immunizationSequenceSelect}>
+            <ComboBox
+              titleText="Vaccines"
+              items={vaccineState}
+              itemToString={(item) => item?.display ?? ''}
+              onChange={(event) => {
+                onChangeComboBox(event);
+              }}
+            />{' '}
+          </section>
+        ) : null}
+
         {hasSequences(formState.sequences) && (
-          <div className={styles.immunizationSequenceSelect}>
```

Comment:
`div` is probably better than `section` for most of these. A `section` is supposed to be part of a document, e.g.,

<section>
  <h2>This is a section</h2>
  <p>And here is some section content</p>
</section>

`div` are just for anything to group a bunch of stuff that need to be grouped like you're doing here.